File: test_integration.py
"""
TruthMate Integration Test Script
Tests the connection between Next.js API and Flask ML service
"""
import requests
import json
import sys

def test_flask_health():
    """Test Flask service health directly"""
    try:
        response = requests.get("http://localhost:5000/health", timeout=5)
        if response.status_code == 200:
            print("✅ Flask ML service is running")
            print(f"   Response: {response.json()}")
            return True
        else:
            print(f"❌ Flask service returned {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ Flask service not reachable: {e}")
        return False

def test_nextjs_health():
    """Test Next.js API health endpoint"""
    try:
        response = requests.get("http://localhost:3000/api/health", timeout=5)
        if response.status_code == 200:
            data = response.json()
            print("✅ Next.js API is running")
            print(f"   ML Service Status: {data['services']['ml_service']}")
            return data['services']['ml_service'] in ['healthy', 'model_loaded']
        else:
            print(f"❌ Next.js API returned {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ Next.js API not reachable: {e}")
        return False

def test_verification_pipeline():
    """Test the complete verification pipeline"""
    try:
        test_claim = "The COVID-19 vaccine is effective at preventing severe illness"
        
        response = requests.post("http://localhost:3000/api/verify", 
                               json={"text": test_claim, "public": True},
                               timeout=15)
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Verification pipeline works!")
            print(f"   Claim: {data.get('claim', 'N/A')}")
            print(f"   Verdict: {data.get('verdict', 'N/A')}")
            print(f"   Confidence: {data.get('confidence', 'N/A')}%")
            print(f"   Verification ID: {data.get('verificationId', 'N/A')}")
            return True
        else:
            print(f"❌ Verification failed with {response.status_code}")
            print(f"   Error: {response.text}")
            return False
    except Exception as e:
        print(f"❌ Verification pipeline error: {e}")
        return False

if __name__ == "__main__":
    print("🧪 TruthMate Integration Test")
    print("=" * 40)
    
    # Test Flask service
    flask_ok = test_flask_health()
    print()
    
    # Test Next.js API
    nextjs_ok = test_nextjs_health()
    print()
    
    # Test full pipeline if both services are running
    if flask_ok and nextjs_ok:
        pipeline_ok = test_verification_pipeline()
        print()
        
        if pipeline_ok:
            print("🎉 All systems working! Your TruthMate integration is ready!")
        else:
            print("⚠️  Services are running but pipeline has issues")
    else:
        print("⚠️  Some services are not running properly")
        if not flask_ok:
            print("   → Start Flask: cd ml-service && python app.py")
        if not nextjs_ok:
            print("   → Start Next.js: npm run dev")
    
    print("\n📋 Next Steps:")
    print("   1. Make sure both services are running")
    print("   2. Visit http://localhost:3000/dashboard")
    print("   3. Submit a claim to test the full pipeline")
    print("   4. Check the results page for AI-powered analysis")

File: test_integration.js
/**
 * Integration test script for TruthMate Platform
 * Tests frontend-backend integration with the ML models
 */

const fetch = require('node-fetch');

const ML_SERVICE_URL = 'http://localhost:5000';
const FRONTEND_API_URL = 'http://localhost:3000/api/verify';

// Test claims for verification
const testClaims = [
    "The Earth is flat",
    "Water boils at 100°C at sea level",
    "COVID-19 vaccines contain microchips",
    "Regular exercise improves cardiovascular health"
];

async function testMLService() {
    console.log("🔬 Testing ML Service Direct Connection...\n");
    
    for (const claim of testClaims) {
        try {
            const response = await fetch(`${ML_SERVICE_URL}/verify`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ text: claim }),
                timeout: 10000
            });

            if (!response.ok) {
                console.log(`❌ ML Service Error for "${claim}": ${response.status}`);
                continue;
            }

            const result = await response.json();
            console.log(`✅ "${claim}"`);
            console.log(`   Verdict: ${result.label} (${result.confidence}% confidence)`);
            console.log(`   Explanation: ${result.explanation.substring(0, 100)}...`);
            console.log("");
        } catch (error) {
            console.log(`❌ ML Service Connection Error for "${claim}": ${error.message}`);
        }
    }
}

async function testFrontendAPI() {
    console.log("🌐 Testing Frontend API Integration...\n");
    
    for (const claim of testClaims) {
        try {
            const response = await fetch(FRONTEND_API_URL, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ 
                    text: claim, 
                    public: true 
                }),
                timeout: 15000
            });

            if (!response.ok) {
                console.log(`❌ Frontend API Error for "${claim}": ${response.status}`);
                const errorText = await response.text();
                console.log(`   Error details: ${errorText}`);
                continue;
            }

            const result = await response.json();
            console.log(`✅ "${claim}"`);
            console.log(`   Verification ID: ${result.verificationId}`);
            console.log(`   Verdict: ${result.verdict} (${result.confidence}% confidence)`);
            console.log(`   Explanation: ${result.explanation.substring(0, 100)}...`);
            console.log(`   Evidence Sources: ${result.evidence?.length || 0}`);
            console.log(`   Harm Index: ${result.harmIndex}`);
            console.log("");
        } catch (error) {
            console.log(`❌ Frontend API Connection Error for "${claim}": ${error.message}`);
        }
    }
}

async function testHealthCheck() {
    console.log("❤️ Running Health Checks...\n");
    
    // Test ML Service Health
    try {
        const mlHealthResponse = await fetch(`${ML_SERVICE_URL}/health`, {
            timeout: 5000
        });
        
        if (mlHealthResponse.ok) {
            const mlHealth = await mlHealthResponse.json();
            console.log("✅ ML Service is healthy");
            console.log(`   Status: ${mlHealth.status}`);
            console.log(`   Models loaded: ${mlHealth.models_loaded}`);
        } else {
            console.log("❌ ML Service health check failed");
        }
    } catch (error) {
        console.log("❌ ML Service is not reachable:", error.message);
    }
    
    // Test Frontend API basic health
    try {
        const frontendResponse = await fetch(`${FRONTEND_API_URL}`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({}),
            timeout: 5000
        });
        
        // We expect a 400 error for empty body, which means the API is responding
        if (frontendResponse.status === 400) {
            console.log("✅ Frontend API is responding");
        } else {
            console.log(`⚠️ Frontend API unexpected response: ${frontendResponse.status}`);
        }
    } catch (error) {
        console.log("❌ Frontend API is not reachable:", error.message);
    }
    
    console.log("");
}

async function runFullIntegrationTest() {
    console.log("🚀 TruthMate Integration Test Suite");
    console.log("=====================================\n");
    
    await testHealthCheck();
    await testMLService();
    await testFrontendAPI();
    
    console.log("🎯 Integration Test Complete!");
    console.log("To start the services manually:");
    console.log("1. ML Service: cd ml-service && python production_sota_service.py");
    console.log("2. Frontend: npm run dev");
}

// Run the tests
if (require.main === module) {
    runFullIntegrationTest().catch(console.error);
}